new_structure/views/auth.py

```
@@ -134,14 +166,45 @@ def teacher_login():
+                'teacher_id': int(os.getenv('SUBJECT_TEACHER2_ID', '5')),
+                'role': 'teacher'
+            }
+        }
+
+        if username in valid_teacher_credentials and password == valid_teacher_credentials[username]['password']:
+            print(f"🔍 Subject teacher authentication successful via bypass mechanism from IP: {client_ip}")
+            session['teacher_id'] = valid_teacher_credentials[username]['teacher_id']
+            session['username'] = username
+            session['role'] = valid_teacher_credentials[username]['role']
             session.permanent = True
+            print(f"🔍 Session set, redirecting to subject teacher dashboard...")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)